programs/psinode/tests/psinode.py

```diff
@@ -298,13 +305,16 @@ def set_producers(self, prods, algorithm=None):
         return self.push_action('producers', 'producers', 'setConsensus', {'consensus':{mode: {'producers': producers}}})
 
     # Queries
-    def graphql(self, service, query):
+    def graphql(self, service, query, token=None):
         '''
         Sends a GraphQL query to a service and returns the result as json
 
         Raise GraphQLError if the query fails
         '''
-        with self.post('/graphql', service=service, json={'query': query}) as result:
+        kwargs = {'json': {'query': query}}
+        if token is not None:
+            kwargs['headers'] = {'Authorization': 'Bearer ' + token}
```

Comment:
I think it would be better always to pass `headers`, but initialize it to an empty dict, instead of using `kwargs` to include or not include `headers`.